src/main/resources/Diagram/MigrationScript.xml

```diff
@@ -0,0 +1,157 @@
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+
+#macro(selectDiagramsRequiringMigration)
+  #set ($repository = $services.extension.getRepository('installed'))
+  #set ($extension = $services.extension.installed.getInstalledExtension('org.xwiki.contrib:draw.io', "wiki:$xcontext.database"))
+  #if ($extension != $NULL)
+    #set ($targetVersion = $extension.version)
+  #end
+  #if ($targetVersion)
+    #set ($migrationIsRequiredClause = " and doc.content like '%draw.io%' and doc.content not like '%draw.io/$targetVersion/%'")
+    #set ($hql = "select doc.fullName from XWikiDocument as doc, BaseObject as obj where obj.name = doc.fullName and obj.className='Diagram.DiagramClass' and doc.fullName &lt;&gt; 'Diagram.DiagramTemplate' $migrationIsRequiredClause")
+    #set ($toBeMigratedEntries = $services.query.hql($hql).execute())
+  #end
+#end
+
+## Display the content below only if the page is not included by another one.
+#if ($doc.name == 'MigrationScript')
+  {{info}}
+  This page contains a script for migrating diagram pages after an upgrade of the diagram application. It will update the paths containing the version of the draw.io library with the version currently used by the application. The script is a temporary solution until the bug //[[Paths to images inserted into diagrams become invalid at upgrade&gt;&gt;https://github.com/xwikisas/application-diagram/issues/11]]// gets fixed.
+  {{html clean="false"}}
+    &lt;form action="" method="post"&gt;
+    &lt;input class="btn btn-default" type="submit" name="execute" value="Execute"/&gt;
+    &lt;input type="hidden" name="form_token" value="$!{services.csrf.token}" /&gt;
+    &lt;/form&gt;
+  {{/html}}
+
+  {{/info}}
+
+  == Diagrams requiring a migration ==
+
+  #selectDiagramsRequiringMigration()
+  {{info}}
+  #if (!$targetVersion)
+   No version of draw.io was found, please make sure that the Diagram Application is installed (either [[pro&gt;&gt;https://store.xwiki.com/xwiki/bin/view/Extension/DiagramApplication]] or [[non-pro&gt;&gt;https://extensions.xwiki.org/xwiki/bin/view/Extension/Diagram%20Application]] version).
+  #else
+    The target draw.io version found is : $targetVersion.
+  #end
+  {{/info}}
+
+  #if ($targetVersion)
+    #set ($columnsProperties = {
+      'doc.location': {"type":"text","size":20,"html":true},
+      'doc.date': {"type":"text","size":10},
+      'doc.author': {"type":"text","size":10,"link":"author"},
+      '_actions': {"sortable":false,"filterable":false,"html":true,"actions":["edit","delete"]}
+    })
+    #set ($options = {
+      'className': 'Diagram.DiagramClass',
+      'resultPage': 'Diagram.MigrationScriptLiveTableResults',
+      'translationPrefix': 'diagram.livetable.',
+      'tagCloud': true,
+      'rowCount': 15,
+      'maxPages': 10,
+      'extraParams': "migrationIsRequiredClause=$escapetool.url($migrationIsRequiredClause)"
```

Comment:
I would move this to ``Diagram.MigrationScriptLiveTableResults``. Listing only the diagrams that require a migration is the main job of ``Diagram.MigrationScriptLiveTableResults`` so it should do it without any extra parameters IMO.

Reply:
In order to do the move you probably need to extract the code that builds ``migrationIsRequiredClause`` into a Velocity macro and put it in a page that can be included by both ``MigrationScript`` and ``MigrationScriptLiveTableResults``.

Reply:
I just realiaze that the "Resolve conversation" action was not taking into account the comment. I thought it was doing "Comment and resolve", so all the comments I added previously were discarded :-). The gist of them was mainly: thank you for helping me out improving the code.